docs/01 - build.md

```diff
@@ -109,17 +129,45 @@ If you want to reuse an existing Kubernetes config, such as your [`EKS`](https:/
-- For development purposes, you will also want to follow the optional steps to install [Helm 3.x](https://helm.sh/docs/intro/install/).
+### An example
+
+Here we present an example of deploying `KubeBlocks` locally(Bring your own toolbox) and utilizing it to create an `apeloud-mysql` cluster.
+
+Before you start, please ensure that you have prior experience with the [basic usage of KubeBlocks](https://kubeblocks.io/docs/preview/user_docs/overview/introduction). You can attempt to create a mysql cluster and establish a connection to it.
+
+#### Deploy KubeBlocks
+- Setup local cluster(Minikube, k3d), e.g., `k3d cluster create mycluster`
+- Install add-on. You can install add-ons through [Helm](https://kubeblocks.io/docs/release-0.8/developer_docs/integration/how-to-add-an-add-on). Here is a simpler example:
+    - Install [`kbcli`](https://kubeblocks.io/docs/release-0.8/user_docs/installation/install-with-kbcli/install-kbcli)
+    - Install [`KubeBlocks`](https://kubeblocks.io/docs/release-0.8/user_docs/installation/install-with-kbcli/install-kubeblocks-with-kbcli): `kbcli kubeblocks install`. This will install `KubeBlocks` and start running its manager.
+    - Install add-ons you need. Please follow guildlines [here](https://kubeblocks.io/docs/preview/user_docs/overview/supported-addons#list-addons). 
+```shell
+# make sure there is an index
+kbcli addon index list
+# search supported addon
+kbcli addon search apecloud-mysql  
+# install addon
+kbcli addon install apecloud-mysql --index kubeblocks --version [YOUR_VERSION]
+# enable addon
+kbcli addon enable apecloud-mysql
+# list the addons again to check whether it is enabled.
+kbcli addon list
+```
+- To deploy KubeBlocks of your version locally, stop the manager of `KubeBlocks` you just install: 
+```shell
+kubectl scale deployment kubeblocks --replicas=0 -n kb-system
+```
+- Generate CRD: `make manifests`. This will create CRD of your KubeBlocks.
+- Deploy CRD: `make install`. This will register CR to cluster.
+- Start your operator: `make run`. This will start your `KubeBlocks` controller and output logs.
+
+#### Test with apecloud-mysql
+- [Create `apecloud-mysql` cluster](https://kubeblocks.io/docs/preview/user_docs/kubeblocks-for-mysql/cluster-management/create-and-connect-a-mysql-cluster) 
+```
+kbcli cluster create apecloud-mysql apecloud-mysql --cluster-definition=apecloud-mysql
+```
+- Now you will see corresponding logs. You can use `kbcli cluster list` to see information of clusters.
+
+#### Debug
+- To observe the behavior of `KubeBlocks` in the example mentioned above, you can set a breakpoint on `Concile` function in `controllers/apps/cluster_controller.go`
+- And then start manager in `cmd/manager/main.go`.
```

Comment:
This paragraph and the previous paragraph are too disjointed; additional content should be added to bridge them.